security/pkg/pki/ra/common.go

```diff
@@ -86,6 +89,45 @@ func ValidateCSR(csrPEM []byte, subjectIDs []string) bool {
 			return false
 		}
 	}
+
+	// Validate the CSR is generated by Istio agent and doesn't have arbitrary fields.
+	// See generate_csr.go GenCSRTemplate and secretcache.go for the counterpart and options
+	goodcsr := &x509.CertificateRequest{}
+	if len(csr.Subject.Organization) > 1 {
+		return false
+	}
+	if len(csr.Subject.Organization) == 1 {
+		goodcsr.Subject.Organization = []string{csr.Subject.Organization[0]}
+	}
+	goodsubj, err := asn1.Marshal(goodcsr.Subject.ToRDNSequence())
+	csrsubj, err := asn1.Marshal(csr.Subject.ToRDNSequence())
```

Comment:
Probably worth mentioning in a comment above that we will have to explicitly set `stub` values in `goodcsr` for each `pki.Name` field we ever add beyond `Organization` here, or this will fail.

Reply:
Also, `k8s_ra_test.go:TestValidateCSR` needs some more test cases, if we intend to be opinionated here.

Reply:
....which is what #51966 adds, so slight pref for that one.